src/selectionHandler.ts

```diff
@@ -0,0 +1,132 @@
+import { Selection, commands, CodeAction, workspace, TextEditor, CodeActionKind } from 'vscode';
+import { tabNineProcess } from './TabNine';
+import * as vscode from 'vscode';
+import { CompletionOrigin } from './extension';
+const importStatement = /Import ([\S]*) from module [\S]*/;
+const existingImportStatement = /Add ([\S]*) to existing import declaration from [\S]*/;
+const importDefaultStatement = /Import default ([\S]*) from module [\S]*/;
+const existingDefaultImportStatement = /Add default import ([\S]*) to existing import declaration from [\S]*/;
+const importStatements = [importStatement, existingImportStatement,importDefaultStatement, existingDefaultImportStatement];
+const DELAY_FOR_CODE_ACTION_PROVIDER = 800;
+
+export const COMPLETION_IMPORTS = 'tabnine-completion-imports';
+
+export async function selectionHandler(editor: TextEditor, edit, {currentCompletion, completions, position }) {
+    try {
+        const eventData = eventDataOf(completions, currentCompletion, editor, position);
+        console.log(eventData);
+        tabNineProcess.setState(eventData);
+
+        handleImports(editor, currentCompletion);
+    } catch (error) {
+        console.error(error);
+    }
+}
+
+function eventDataOf(completions: any, currentCompletion: any, editor: TextEditor, position: any) {
+    let index = completions.findIndex(({new_prefix}) => new_prefix == currentCompletion);
+
+    let numOfVanillaSuggestions = 0;
+    let numOfDeepLocalSuggestions = 0;
+    let numOfDeepCloudSuggestions = 0;
+    let numOfLspSuggestions = 0;
+    let currInCompletions = completions[index];
+
+    let suggestions = completions.map(c => {
+        if (c.origin == CompletionOrigin.VANILLA) {
+            numOfVanillaSuggestions += 1;
+        } else if (c.origin == CompletionOrigin.LOCAL) {
+            numOfDeepLocalSuggestions += 1;
+        } else if (c.origin == CompletionOrigin.CLOUD) {
+            numOfDeepCloudSuggestions += 1;
+        } else if (c.origin == CompletionOrigin.LSP) {
+            numOfLspSuggestions += 1;
+        }
+
+        return { 
+            length: c.new_prefix.length, 
+            strength: resolveDetailOf(c),
+            origin: c.origin
+        };
+    });
+
+    const length = currentCompletion.length;
+    const netLength = editor.selection.anchor.character - position.character;
+    const strength = resolveDetailOf(currInCompletions);
+    const origin = currInCompletions.origin;
+    const prefixLength = editor.document.getText(new vscode.Range(new vscode.Position(position.line, 0), position)).trimLeft().length;
+    const netPrefixLength = prefixLength - (currentCompletion.length - netLength);
+    const language = editor.document.fileName.split('.').pop();
```

Comment:
there is a `language` property `document.languageId` you can use it to identify the language

Reply:
is it the file's extension?
because if not i cant use it